app/jobs/get_usps_proofing_results_job.rb

```diff
@@ -111,25 +111,34 @@ def check_enrollments(enrollments)
       status_check_attempted_at = Time.zone.now
       enrollment_outcomes[:enrollments_checked] += 1
       response = nil
-      errored = true
 
       begin
         response = proofer.request_proofing_results(
           enrollment.unique_id, enrollment.enrollment_code
         )
-        errored = false
       rescue Faraday::BadRequestError => err
+        # 400 status code. This is used for some status updates and some common client errors
         handle_bad_request_error(err, enrollment)
+      rescue Faraday::ClientError, Faraday::ServerError => err
+        # 4xx or 5xx status code. These are unexpected but will have some sort of
+        # response body that we can try to log data from
+        NewRelic::Agent.notice_error(err)
+        handle_client_or_server_error(err, enrollment)
+      rescue Faraday::Error => err
+        # Timeouts, failed connections, parsing errors, and other HTTP errors. These
+        # generally won't have a response body
+        NewRelic::Agent.notice_error(err)
+        handle_faraday_error(err, enrollment)
       rescue StandardError => err
         NewRelic::Agent.notice_error(err)
         handle_standard_error(err, enrollment)
+      else
+        process_enrollment_response(enrollment, response)
+      ensure
+        # Record the attempt to update the enrollment
+        enrollment.update(status_check_attempted_at: status_check_attempted_at)
       end
 
-      process_enrollment_response(enrollment, response) unless errored
```

Comment:
TIL about the `else` clause in Ruby begin/rescue blocks. I switched over to using that since it's simpler